File: Examples/test-suite/java/director_string_runme.java

import director_string.*;

public class director_string_runme {

  static {
    try {
      System.loadLibrary("director_string");
    } catch (UnsatisfiedLinkError e) {
      System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e);
      System.exit(1);
    }
  }

  public static void main(String argv[]) {

    director_string_B b = new director_string_B("hello");

    String s;

    s = b.call_get_first();
    if (!s.equals("director_string_B.get_first")) throw new RuntimeException("call_get_first() failed");

    s = b.call_get(0);
    if (!s.equals("director_string_B.get")) throw new RuntimeException("get(0) failed");
//    if (!s.equals("director_string_B.get: hello")) throw new RuntimeException("get(0) failed");
  }
}

class director_string_B extends A {
    public director_string_B(String first) {
      super(first);
    }
    public String get_first() {
      return "director_string_B.get_first";
    }
  
    public String get(int n) {
      return "director_string_B.get";
// recursive call problem (needs fixing)
//      return "director_string_B.get: " + super.get(n);
    }
}

File: Examples/test-suite/kotlin/director_string_runme.kt
@file:JvmName("director_string_runme")

import director_string.*

class director_string_B(first: String) : A(first) {
    override fun get_first(): String {
        return "director_string_B.get_first"
    }

    override fun get(n: Int): String {
        return "director_string_B.get: " + super.get(n)
    }
}

class director_string_A(first: String) : A(first) {
    override fun get(n: Int): String {
        return Integer.valueOf(n).toString()
    }
}

fun main() {
    try {
        System.loadLibrary("director_string")
    } catch (e: UnsatisfiedLinkError) {
        System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e)
        kotlin.system.exitProcess(1)
    }

    var s: String

    val c = director_string_A("hi")
    for (i in 0..2) {
        s = c.call_get(i)
        if (s != Integer.valueOf(i).toString()) throw RuntimeException("director_string_A.get(" + i + ") failed. Got:" + s)
    }

    val b = director_string_B("hello")

    s = b.get_first()
    if (s != "director_string_B.get_first") throw RuntimeException("get_first() failed")

    s = b.call_get_first()
    if (s != "director_string_B.get_first") throw RuntimeException("call_get_first() failed")

    s = b.call_get(0)
    if (s != "director_string_B.get: hello") throw RuntimeException("get(0) failed")
}
